.github/workflows/maven.yml

```diff
@@ -7,19 +7,64 @@ on:
     branches: ["main"]
 
 jobs:
-  build:
+  build-project:
     runs-on: ubuntu-latest
 
     steps:
-    - name: Checkout code
-      uses: actions/checkout@v4
-
-    - name: Set up JDK 17
-      uses: actions/setup-java@v3
-      with:
-        java-version: '17'
-        distribution: 'temurin'
-        cache: maven
-
-    - name: Build and check coverage
-      run: mvn -B verify --file pom.xml
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+
+      - name: Build and check coverage
+        run: mvn -B clean install --file pom.xml
+
+      - name: Upload JAR artifact
```

Comment:
zastanawiam sie czy nie powinnismy zbudowac jara juz w nastepnym job jesli tylko tam tego uzywamy i tak

Reply:
Imo niepotrzebnie. Ten mvn clean install robi ci jara plus coverage checki więc imo warto to wykorzystać. Tak to operacja będzie dwa razy dłużej trwać a i tak zrobi te same dwa jary

Reply:
A to chyba że tak

Reply:
To można mergowac chyba